Shared/Views/DonorChart.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  DonorChart.swift
+//  St Jude
+//
+//  Created by Ben Cardy on 17/09/2022.
+//
+
+import SwiftUI
+import Charts
+
+struct DonorChartValue: Identifiable {
+    let id: Int
+    let amount: Double
+    let date: Date
+}
+
+struct DonorChart: View {
+    
+    let donations: [TiltifyDonorsForCampaignDonation]
+    let total: TiltifyAmount
+    
+    @State private var maxValue: Double = 0
+    @State private var minValue: Double = 0
+    @State private var showDonationValues: Bool = false
+    @State private var chartValues: [DonorChartValue] = []
+    
+    var body: some View {
+        if #available(iOS 16.0, *) {
+            Chart {
+                ForEach(chartValues) { donation in
+                    if showDonationValues {
+                        BarMark(
+                            x: .value("Date", donation.date),
+                            y: .value("Donation Amount", donation.amount),
+                            width: 2
+                        )
+                    } else {
+                        LineMark(
+                            x: .value("Date", donation.date),
+                            y: .value("Donation Amount", donation.amount)
+                        )
+                    }
+                }
+            }
+            .chartYScale(domain: ClosedRange(uncheckedBounds: (minValue, maxValue)))
+            .chartYAxis(.hidden)
+            //            .chartXAxis(.hidden)
+            .onTapGesture {
+                showDonationValues.toggle()
+            }
+            .onAppear {
+                calculateChartValues()
+            }
+            .onChange(of: showDonationValues) { newValue in
+                calculateChartValues()
+            }
+        } else {
+            Text("Sorry!")
+        }
+    }
+    
+    func calculateChartValues() {
+        let reversedDonations = donations.filter {
```

Comment:
Are they reversed?

Reply:
Yeah, in the API and the list they're "most recent first" but for the graph we need "oldest first"

Reply:
But the filter operation doesn't reverse them?